src/overlays/actors/ovl_Boss_Fd/z_boss_fd.c

```diff
@@ -23,275 +57,1937 @@ const ActorInit Boss_Fd_InitVars = {
+            this->targetPosition.x = sCeilingTargets[this->work[BFD_CEILING_TARGET]].x;
+            this->targetPosition.y = sCeilingTargets[this->work[BFD_CEILING_TARGET]].y + 900.0f;
+            this->targetPosition.z = sCeilingTargets[this->work[BFD_CEILING_TARGET]].z;
+            if (this->timers[0] == 0) {
+                this->timers[0] = 25;
+                this->work[BFD_CEILING_TARGET]++;
+                if (this->work[BFD_CEILING_TARGET] >= 6) {
+                    this->work[BFD_CEILING_TARGET] = 0;
+                }
+            }
+            Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 50.0f, 50.0f, 100.0f, 2);
+            if (this->timers[1] == 0) {
+                osSyncPrintf("BGCHECKKKKKKKKKKKKKKKKKKKKKKK\n");
+                if (this->actor.bgCheckFlags & 0x10) {
+                    this->fwork[BFD_CEILING_BOUNCE] = -18384.0f;
+                    this->timers[1] = 10;
+                    Audio_PlaySoundGeneral(NA_SE_EV_EXPLOSION, &this->actor.projectedPos, 4, &D_801333E0, &D_801333E0,
+                                           &D_801333E8);
+                    func_80033E1C(globalCtx, 3, 0xA, 0x7530);
+                    this->work[BFD_ROCK_TIMER] = 300;
+                }
+            } else {
+                pitchToTarget = this->fwork[BFD_CEILING_BOUNCE];
+                Math_ApproachZeroF(&this->fwork[BFD_CEILING_BOUNCE], 1.0f, 1000.0f);
+            }
+            if (this->timers[2] == 0) {
+                this->work[BFD_ACTION_STATE] = BOSSFD_FLY_MAIN;
+                this->timers[0] = 0;
+                this->work[BFD_START_ATTACK] = false;
+            }
+            break;
+        case BOSSFD_FLY_CHASE:
+            this->actor.flags |= 0x1000000;
+            temp_y = Math_SinS(this->work[BFD_MOVE_TIMER] * 2396.0f) * 30.0f + this->fwork[BFD_TARGET_Y_OFFSET];
+            this->targetPosition.x = player->actor.world.pos.x;
+            this->targetPosition.y = player->actor.world.pos.y + temp_y + 30.0f;
+            this->targetPosition.z = player->actor.world.pos.z;
+            this->fwork[BFD_FLY_WOBBLE_AMP] = 0.0f;
+            if (((this->timers[0] % 64) == 0) && (this->timers[0] < 450)) {
+                this->work[BFD_ROAR_TIMER] = 40;
+                if (BossFd_IsFacingLink(this)) {
+                    this->fireBreathTimer = 20;
+                }
+            }
+            if ((this->work[BFD_DAMAGE_FLASH_TIMER] != 0) || (this->timers[0] == 0) ||
+                (player->actor.world.pos.y < 70.0f)) {
+                this->work[BFD_ACTION_STATE] = BOSSFD_FLY_MAIN;
+                this->timers[0] = 0;
+                this->work[BFD_START_ATTACK] = false;
+            } else {
+                Math_ApproachF(&this->fwork[BFD_TARGET_Y_OFFSET], 50.0, 1.0f, 2.0f);
+            }
+            break;
+        case 199: // This makes up for the missing float in rodata so it doesn't shift the whole ROM
+        case BOSSFD_DEATH_START:
```

Comment:
wait so the case isnt real, and youre just adding it because the float is missing? confused. if this doesnt exist it should just not be there. if you build non matching you already know youre gonna get a shifted rom

Reply:
It really helps when testing for matching. But I suppose it's probably better to leave out.